db/revision_cache_lru.go

```diff
@@ -539,13 +535,22 @@ func (rc *LRURevisionCache) revCacheMemoryBasedEviction() {
 
 // performEviction will evict the oldest items in the cache till we are below the memory threshold
 func (rc *LRURevisionCache) performEviction() {
-	rc.lock.Lock()
-	defer rc.lock.Unlock()
-	var numItemsRemoved int64
-	for rc.currMemoryUsage.Value() > rc.memoryCapacity {
-		rc.purgeOldest_()
-		numItemsRemoved++
+	var numItemsRemoved, numBytesRemoved int64
+	rc.lock.Lock() // hold rev cache lock to remove items from cache until we're below memory threshold for the shard
+	// check if we are over memory capacity after holding rev cache mutex (protect against another goroutine evicting whilst waiting for mutex above)
+	if rc.currMemoryUsage.Value() > rc.memoryCapacity {
+		// find amount of bytes needed to evict till below threshold
+		bytesNeededToRemove := rc.currMemoryUsage.Value() - rc.memoryCapacity
```

Comment:
Just avoids a duplicate atomic load

```suggestion
	if currMemoryUsage := rc.currMemoryUsage.Value(); currMemoryUsage > rc.memoryCapacity {
		// find amount of bytes needed to evict till below threshold
		bytesNeededToRemove := currMemoryUsage - rc.memoryCapacity
```